test/e2e/performance_test.go

```diff
@@ -150,11 +150,20 @@ func setupTestPodsConnection(data *TestData) error {
 func generateWorkloadNetworkPolicy(policyRules int) *networkv1.NetworkPolicy {
 	ingressRules := make([]networkv1.NetworkPolicyPeer, policyRules)
 	// #nosec G404: random number generator not used for security purposes
-	rnd := rand.New(rand.NewSource(seed))
-	existingCIDRs := make(map[string]struct{}) // ensure no duplicated cidrs
+
+	// Initialize the PCG generator
+	pcg := rand.PCG{}
+	// Provide both 'state' and 'sequence' seeds
+	pcg.Seed(uint64(seed), uint64(1))
+	rnd := rand.New(&pcg)
```

Comment:
why are we using this verbose code here instead of `rand.New(rand.NewPCG(seed, 0))` like in the rest of the PR?